Starwatch.Core/Monitoring/AnnouncementMonitor.cs

```diff
@@ -40,15 +40,23 @@ public class Announcement
         public bool EnableMonitor { get; set; }
 
         public Announcement[] Announcements = new Announcement[0];
+
         public Timer[] AnnouncementTimers = new Timer[0];
 
+        private bool Locked = false;
+
         public AnnouncementMonitor(Server server) : base(server, "AnnouncementMonitor")
         {
             Logger.Colourise = true;
```

Comment:
Generally speaking, good idea to let colourise be determined by the system configuration. It can actually slow down the process since colouring is pretty expensive.

Reply:
Yeah, I'm not familiar enough with semaphores or lock to understand what to do here to improve it. It's used for the route so hopefully no one should be spamming it enough to cause a slow down.

Reply:
Sorry, Github highlighted the wrong line. This comment was pointed at line 50: `Logger.Colourise`